File: Tests/Main Tests/ExcelAddinCSharp/SampleControl.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace ExcelAddinCSharp
{
    public partial class SampleControl : UserControl, NetOffice.ExcelApi.Tools.ITaskPane
    {
        public SampleControl()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            MessageBox.Show("Hello");
        }

        void NetOffice.ExcelApi.Tools.ITaskPane.OnConnection(NetOffice.ExcelApi.Application application, NetOffice.OfficeApi._CustomTaskPane parentPane, object[] customArguments)
        {
            TestAddin addin = customArguments[0] as TestAddin;
            addin.TaskPaneOkay = true;
        }

        void NetOffice.ExcelApi.Tools.ITaskPane.OnDisconnection()
        {
             
        }

        void NetOffice.ExcelApi.Tools.ITaskPane.OnDockPositionChanged(NetOffice.OfficeApi.Enums.MsoCTPDockPosition position)
        {
           
        }

        void NetOffice.ExcelApi.Tools.ITaskPane.OnVisibleStateChanged(bool visible)
        {

        }
    }
}

File: Tests/Main Tests/ExcelAddinVB/SampleControl.vb
Public Class SampleControl
    Implements NetOffice.ExcelApi.Tools.ITaskPane

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        MsgBox("Hello")
    End Sub

    Public Sub OnConnection(ByVal application As NetOffice.ExcelApi.Application, ByVal parentPane As NetOffice.OfficeApi._CustomTaskPane, ByVal customArguments() As Object) Implements NetOffice.ExcelApi.Tools.ITaskPane.OnConnection

        Dim addin As TestAddin = customArguments(0)
        addin.TaskPaneOkay = True

    End Sub

    Public Sub OnDisconnection() Implements NetOffice.ExcelApi.Tools.ITaskPane.OnDisconnection

    End Sub

    Public Sub OnDockPositionChanged(ByVal position As NetOffice.OfficeApi.Enums.MsoCTPDockPosition) Implements NetOffice.ExcelApi.Tools.ITaskPane.OnDockPositionChanged

    End Sub


    Public Sub OnVisibleStateChanged(ByVal visible As Boolean) Implements NetOffice.ExcelApi.Tools.ITaskPane.OnVisibleStateChanged

    End Sub

End Class
